compiler/rustc_next_trait_solver/src/solve/eval_ctxt/mod.rs

```diff
@@ -1065,14 +1078,13 @@ where
         &mut self,
         key: ty::OpaqueTypeKey<I>,
     ) -> Option<(ty::OpaqueTypeKey<I>, I::Ty)> {
-        let mut matching =
-            self.delegate.clone_opaque_types_for_query_response().into_iter().filter(
-                |(candidate_key, _)| {
-                    candidate_key.def_id == key.def_id
-                        && DeepRejectCtxt::relate_rigid_rigid(self.cx())
-                            .args_may_unify(candidate_key.args, key.args)
-                },
-            );
+        let mut matching = self.delegate.clone_opaque_types_lookup_table().into_iter().filter(
```

Comment:
Either iterate over the duplicates or leave a comment (or FIXME) that we probably should, but don't have a good reason to do so yet.